charts/orchestrator/templates/openshift-rhsso.yaml

```diff
@@ -0,0 +1,158 @@
+kind: Keycloak
+metadata:
+  name: {{ .Values.rhsso.realm.name }}
+  namespace: {{ .Values.rhsso.namespace }}
+  labels:
+   app: {{ $appLabel }}
+spec:
+  keycloakDeploymentSpec:
+    experimental:
+      env:
+      - name: KEYCLOAK_LOGLEVEL
+        value: {{ .Values.rhsso.log_level }}
+      - name: KC_LOG_LEVEL
+        value: {{ .Values.rhsso.log_level }}
+  instances: {{ .Values.rhsso.instances }}
+  externalAccess:
+    enabled: True
+---
+apiVersion: keycloak.org/v1alpha1
+kind: KeycloakRealm
+metadata:
+  name: {{ .Values.rhsso.realm.name }}
+  namespace: {{ .Values.rhsso.namespace }}
+  labels:
+    app: {{ $appLabel }}
+spec:
+  realm:
+    id: {{ .Values.rhsso.realm.name }}
+    realm: {{ .Values.rhsso.realm.name }}
+    enabled: True
+    displayName: "Backstage realm"
+    roles:
+      realm:
+      - clientRole: false
+        composite: true
+        composites:
+          client:
+            realm-management:
+            - query-users
+            - query-clients
+            - view-users
+            - query-groups
+        containerId: backstage
+        description: Access from backstage
+        name: priv-client
+  instanceSelector:
+    matchLabels:
+      app: {{ $appLabel }}
+---
+apiVersion: keycloak.org/v1alpha1
+kind: KeycloakClient
+metadata:
+  labels:
+    app: {{ $appLabel }}
+  name: {{ .Values.rhsso.realm.name }}
+  namespace: {{ .Values.rhsso.namespace }}
+spec:
+  client:
+    enabled: true
+    clientAuthenticatorType: client-secret
+    clientId: {{ .Values.rhsso.client.username }}
+    name: {{ .Values.rhsso.client.username }}
+    serviceAccountsEnabled: true
+    redirectUris:
+      - "*"
+    clientId: {{ .Values.rhsso.client.username }}
+    secret: {{ .Values.rhsso.client.password }}
+    defaultClientScopes:
+    - web-origins
+    - profile
+    - email
+    - roles
+    - address
+    - phone
+    - offline_access
+    implicitFlowEnabled: true
+    publicClient: false
+    standardFlowEnabled: true
+    fullScopeAllowed: true
+    directAccessGrantsEnabled: true
+    consentRequired: false
+    protocolMappers:
+    - config:
+        access.token.claim: "true"
+        claim.name: sub
+        id.token.claim: "true"
+        jsonType.label: String
+        user.attribute: username
+        userinfo.token.claim: "true"
+      name: backstage-uid
+      protocol: openid-connect
+      protocolMapper: oidc-usermodel-property-mapper
+  serviceAccountRealmRoles:
+  - priv-client
+  realmSelector:
+    matchLabels:
+      app: {{ $appLabel }}
+
+{{- range .Values.rhsso.users }}
+---
+apiVersion: keycloak.org/v1alpha1
+kind: KeycloakUser
+metadata:
+  name: {{ .name }}
+  namespace: {{ $.Values.rhsso.namespace }}
+spec:
+  realmSelector:
+    matchLabels:
+      app: {{ $appLabel }}
+  user:
+    email: {{ .email }}
+    emailVerified: true
+    enabled: true
+    firstName: {{ .name }}
+    username: {{ .name }}
+    realmRoles:
+      - offline_access
+    credentials:
```

Comment:
Any credentials or user identity can't be managed by the chart/helm-operator because they are visible in the summary of the deployment.

Reply:
how do you manage this? credential is a string cannot be encoded into a secret :\

Reply:
All I can think of is that the CR can't be managed by helm. @masayag WDYT? If this is included in the chart, I'm concerned the security team will flag this as a risk as the CR status will show in base64 encoded the contents of this CR. Anyone looking for this information and able to list the CR will be able to retrieve this value without much effort.

Reply:
Alternatively, moving to a go based operator will solve this issue. @pkliczewski  Is this feature supposed to be included in 1.2/m2?

Reply:
I looked at https://github.com/keycloak/keycloak-realm-operator/blob/f0b2224aee10685b49c48febf38ddfbbe2e6ba0b/pkg/apis/keycloak/v1alpha1/keycloakclient_types.go#L80 and don't see an option to escape it.

Reply:
We can't add it to the chart or to our CR as a field. Besides what keycloak does, I would think that these kind of values would be stored in an object that can be managed by a vault/secret manager. I don't know if this is possible already with Keycloak, but what is for sure is that we can't have it in our chart as a value. 
The only way forward is to deploy this outside the orchestrator CR.